Libs/RaptureCore/src/test/java/rapture/kernel/scripting/ReflexParamsTest.java

```diff
@@ -44,89 +46,139 @@
 
-    private String uri = "//reflexParamsTest";
-    private String scr = "thisDoc1";
+    private String auth = "reflexParamsTest";
+    RaptureURI uri;
 
     @After
     public void cleanUp() {
-        Kernel.getDoc().deleteDocRepo(ctx, uri);
+        Kernel.getDoc().deleteDocRepo(ctx, uri.toAuthString());
     }
 
     @Before
     public void setup() {
         Kernel.initBootstrap();
-        if (!Kernel.getDoc().docRepoExists(ctx, uri)) {
-            Kernel.getDoc().createDocRepo(ctx, uri, "NREP {} USING MEMORY {}");
+        uri = new RaptureURI(auth, Scheme.SCRIPT);
+
+        if (!Kernel.getDoc().docRepoExists(ctx, uri.toAuthString())) {
+            Kernel.getDoc().createDocRepo(ctx, uri.toAuthString(), "NREP {} USING MEMORY {}");
         }
-        if (Kernel.getScript().doesScriptExist(ctx, uri + "/" + scr)) {
-            Kernel.getScript().deleteScript(ctx, uri + "/" + scr);
+    }
+
+    public String makeScript(String name, String script) {
+        String scriptUri = RaptureURI.builder(uri).docPath(name).asString();
+        
+        if (Kernel.getScript().doesScriptExist(ctx, scriptUri)) {
+            Kernel.getScript().deleteScript(ctx, scriptUri);
         }
 
-        Kernel.getScript().createScript(
-                ctx,
-                uri + "/" + scr,
-                RaptureScriptLanguage.REFLEX,
-                RaptureScriptPurpose.PROGRAM,
-                "if defined(_params) do \n" + "  println(\"params is defined \" + _params); return \"def\";\n" + "else do \n"
-                        + "  println(\"params is undefined\"); return \"notdef\";\n" + "end");
+        Kernel.getScript().createScript(ctx, scriptUri, RaptureScriptLanguage.REFLEX, RaptureScriptPurpose.PROGRAM, script);
+        return scriptUri;
     }
 
     @Test
     public void testParamsNotDefined() {
-        String retval = Kernel.getScript().runScript(ctx, uri + "/" + scr, new HashMap<String, String>());
+        String scriptUri = makeScript("testScript1", "if defined(_params) do \n" + "  println(\"params is defined \" + _params); return \"def\";\n"
+                + "else do \n" + "  println(\"params is undefined\"); return \"notdef\";\n" + "end");
+        String retval = Kernel.getScript().runScript(ctx, scriptUri, new HashMap<String, String>());
         assertEquals("notdef", retval);
     }
 
+    // @Test
+    // public void testRAP4090() {
+    // // String scriptUri = makeScript("rap4090", "if defined(web['foo']) do \n" + " println(\"web['foo'] is defined \" + web['foo']); return \"def\";\n"
+    // // + "else do \n" + " println(\"web['foo'] is undefined\"); return \"notdef\";\n" + "end");
+    // String scriptUri = makeScript("rap4090", "a = defined(web['foo']);");
+    // String retval = Kernel.getScript().runScript(ctx, scriptUri, new HashMap<String, String>());
+    // assertEquals("notdef", retval);
+    // }
+
+    @Test
+    public void testRAP4090a() {
```

Comment:
you should name your tests with what you are actually testing.  people looking at error logs dont want to go to jira